src/Sdk/StrixMusic.Sdk/AdapterModels/MergedImage.cs

```diff
@@ -70,7 +74,10 @@ public void RemoveSource(ICoreImage itemToRemove)
         /// <inheritdoc/>
         public bool Equals(ICoreImage other)
         {
-            return other?.Uri == Uri;
+            // We can't know for sure if 2 images are the same until we open the streams and compare them,
+            // which is an asynchronous operation and can't be done here.
+            // For merging, this check can be done as we retrieve the images from the collection.
+            return false;
```

Comment:
Then don't return false, instead throw an exception, maybe

Reply:
We can return ReferenceEquals instead of False here.